### PR TITLE
bugfix: return error when iam key or secret are not present

### PR DESCRIPTION
**Pull request by susana-garcia**
bugfix: return an error when iam key or secret are not present when setting connection details

## Checklist

- [X] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [X] PR contains a single logical change (to build a better changelog).
- [X] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: operator/iamkeycontroller/pipeline.go [hunk 1]
```diff
@@ -61,11 +63,17 @@ func NewPipeline(client client.Client, recorder event.Recorder, exoscaleClient *
 	}
 }
 
-func toConnectionDetails(iamKey *exoscalesdk.IAMAccessKey) managed.ConnectionDetails {
+func toConnectionDetails(iamKey *exoscalesdk.IAMAccessKey) (managed.ConnectionDetails, error) {
+	if iamKey.Key == nil {
+		return nil, fmt.Errorf("iamKey key not found in connection details")
```
Comment (mweibel) [inline operator/iamkeycontroller/pipeline.go:68]:
```suggestion
		return nil, errors.New("iamKey key not found in connection details")
```

`fmt.Errorf` is used when you need a format string in the error which gets replaced by something (usually `%w` which wraps an error). If you just need a normal error value, `errors.New()` is the way to go.

File: operator/iamkeycontroller/pipeline.go [hunk 2]
```diff
@@ -61,11 +63,17 @@ func NewPipeline(client client.Client, recorder event.Recorder, exoscaleClient *
...
+	}
+	if iamKey.Secret == nil {
+		return nil, fmt.Errorf("iamKey secret not found in connection details")
```
Comment (mweibel) [inline operator/iamkeycontroller/pipeline.go:71]:
```suggestion
		return nil, errors.New("iamKey secret not found in connection details")
```

**Review by mweibel — COMMENTED** [review]
slightly nit comment but might be good to fix. Otherwise looks good :+1:

File: operator/iamkeycontroller/pipeline.go (hunk 1 above)
Comment (susana-garcia) [reply operator/iamkeycontroller/pipeline.go]: true that, cheers

**Review by mweibel — APPROVED** [review]

**Review by glrf — APPROVED** [review]
LGTM, should fix the panic

**Pull request merged**